src/ffi/types.rs

```diff
@@ -38,6 +38,27 @@ pub enum TelioError {
     NotStarted,
 }
 
+impl TelioError {
+    pub fn kind(&self) -> String {
+        match self {
+            TelioError::UnknownError(_) => "UnknownError".to_owned(),
+            TelioError::InvalidKey => "InvalidKey".to_owned(),
+            TelioError::BadConfig => "BadConfig".to_owned(),
+            TelioError::LockError => "LockError".to_owned(),
+            TelioError::InvalidString => "InvalidString".to_owned(),
+            TelioError::AlreadyStarted => "AlreadyStarted".to_owned(),
+            TelioError::NotStarted => "NotStarted".to_owned(),
+        }
+    }
+
+    pub fn message(&self) -> String {
+        match self {
+            TelioError::UnknownError(inner) => inner.to_string(),
```

Comment:
nit: Is it possible that the `inner.to_string()` will result in the same string as the `self.kind()`? Maybe it would be safer to add some sort of prefix ("Unknown error: ") to distinguish it?